src/test/java/com/udojava/evalex/TestLazyIf.java

```diff
@@ -26,7 +26,23 @@ public void testLazyIfWithNestedFunction() {
         expression.setVariable("a", new BigDecimal(0));
         Assert.assertEquals(expression.eval(),new BigDecimal(0));
     }
-    
+
+  @Test
+  public void testLazyIfWithNested() {
+    Expression expression = new Expression("if(a=0,0,if(5/a>3,2,4))");
+    expression.setVariable("a", new BigDecimal(0));
+    Assert.assertEquals(expression.eval(),new BigDecimal(0));
+  }
+
+  @Test(expected = ArithmeticException.class)
+  public void testLazyIfWithNested2() {
```

Comment:
You should name it something like `testLazyIfWithNestedFailingIf` to make it clear that this test is supposed to be the test of a failure case.

Reply:
Hello @RobertZenz, thanks for your review. I made changes